@here/harp-geoutils/lib/coordinates/GeoPolygon.ts

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright (C) 2017-2020 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { Vector2 } from "three";
+
+import { GeoBox } from "./GeoBox";
+import { GeoCoordinates, MAX_LATITUDE, MIN_LATITUDE } from "./GeoCoordinates";
+import { GeoCoordinatesLike } from "./GeoCoordinatesLike";
+import { GeoCoordLike, geoCoordLikeToGeoCoordinatesLike } from "./GeoCoordLike";
+import { GeoPolygonLike } from "./GeoPolygonLike";
+
+type MinThreeItemsArray<T> = [T, T, T, ...T[]];
+
+export type GeoPolygonCoordinates = MinThreeItemsArray<
+    GeoCoordinatesLike | GeoCoordinates | GeoCoordLike
+>;
+
+/**
+ * A GeoPolygon in 2D Space (altitudes will be ignored).
+ * Coordinates are expected in counter-clockwise order, for concave polygons a sorting is
+ * available.
+ * Clockwise ordered or selfintersecting Polygons might lead to no or unexpected results.
+ *
+ * @beta @internal
+ */
+export class GeoPolygon implements GeoPolygonLike {
+    coordinates: GeoPolygonCoordinates;
+
+    /**
+     * Creates a GeoPolygon instance
+     *
+     * @param coordinates An array of GeoCoordinates acting as the Vertices of the Polygon.
+     * @param needsSort  If `true` it will sort the coordinates in ccw order, this will only
+     *  result correctly for convex polygons @default false
+     */
+    constructor(coordinates: GeoPolygonCoordinates, needsSort: boolean = false) {
+        this.coordinates = coordinates;
+        if (needsSort) {
+            this.sortCCW();
+        }
+    }
+
+    /**
+     * Gets a BoundingBox for the Polygon
+     */
+    getGeoBoundingBox(): GeoBox | undefined {
```

Comment:
As a user I would not expect this method to return undefined but rather an empty box if the area is `0`

Reply:
sorry, to not return undefined of course

Reply:
The Area is mostly 0, not because the Polygon has no extend, but because the Polygon is twisted or in some complex concave form. But right, the cases should be distinguished

Reply:
done